### PR TITLE
Add TimeVariant wrapper

### PR DESCRIPTION
**Pull request by Manishearth**
Fixes https://github.com/unicode-org/icu4x/issues/1857

### CODE REVIEW

File: components/timezone/src/types.rs [hunk 1]
```diff
@@ -123,3 +126,51 @@ impl FromStr for GmtOffset {
...
     }
 }
+
+/// A time variant, e.g. "daylight" or "standard"
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash, ULE)]
+#[repr(transparent)]
+#[cfg_attr(feature = "datagen", derive(serde::Serialize))]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[allow(clippy::exhaustive_structs)] // newtype
+pub struct TimeVariant(pub TinyStr8);
```
Comment (sffc) [inline components/timezone/src/types.rs:136]: Thought: We should consider making this a `TinyStr4`

**Review by sffc — APPROVED** [review]

File: components/timezone/src/types.rs (hunk 1 above)
Comment (Manishearth) [reply components/timezone/src/types.rs:136]: and shortening the string? could work!

Comment (Manishearth) [reply components/timezone/src/types.rs:136]: `"std"` and `"day"` seem nice (or `"dst"`)

Comment (sffc) [reply components/timezone/src/types.rs:136]: `"std"` and `"dst"` sgtm. And for a future Ramadan variant we could do `"rmn"` or something.

Maybe even make it a `TinyAsciiStr<3>` :)

Comment (sffc) [reply components/timezone/src/types.rs:136]: https://github.com/unicode-org/icu4x/issues/2292

**Pull request merged**